### PR TITLE
chore(modules-sdk,orchestration): to remote joiner query

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
Helper function to transform js/json object into RemoteJoinerQuery format.

```typescript
toRemoteJoinerQuery({
  product: {
    fields: ["id", "title"],
    __args: {
      skip: 0,
    },
    variants: {
      fields: ["id", "title", "handle", "sku"],
      shipping_profile: {
        profile: {
          fields: ["id", "name"],
        },
      },
    },
    collections: {
      fields: ["id", "title"],
    },
  },
})
```

outputs:
```
{
  alias: "product",
  fields: ["id", "title"],
  expands: [
    {
      property: "product.variants",
      fields: ["id", "title", "handle", "sku"],
    },
    {
      property: "product.variants.shipping_profile",
    },
    {
      property: "product.variants.shipping_profile.profile",
      fields: ["id", "name"],
    },
    {
      property: "product.collections",
      fields: ["id", "title"],
    },
  ],
  args: [
    {
      name: "skip",
      value: 0,
    },
  ],
};
```

### CODE REVIEW

File: packages/orchestration/src/joiner/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+export function toRemoteJoinerQuery(obj: any): RemoteJoinerQuery {
+  const remoteJoinerQuery: RemoteJoinerQuery = {
+    alias: "",
+    fields: [],
+    expands: [],
+  }
+
+  let isEntryPoint = true
+  function extractRecursive(obj, parentName = "") {
+    for (let key in obj) {
```
Comment (adrien2p) [inline packages/orchestration/src/joiner/helpers.ts]: suggestion: const

File: packages/orchestration/src/joiner/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      const value = obj[key]
+
+      const canExpand =
+        typeof value === "object" &&
```
Comment (adrien2p) [inline packages/orchestration/src/joiner/helpers.ts:15]: suggestion: isObject?

File: packages/orchestration/src/joiner/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { RemoteJoinerQuery } from "@medusajs/types"
+
+export function toRemoteJoinerQuery(obj: any): RemoteJoinerQuery {
+  const remoteJoinerQuery: RemoteJoinerQuery = {
+    alias: "",
+    fields: [],
+    expands: [],
+  }
+
+  let isEntryPoint = true
```
Comment (adrien2p) [inline packages/orchestration/src/joiner/helpers.ts]: suggestion: can make it the third args of extractRecursive with true as a default value

File: packages/orchestration/src/joiner/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        reference.fields = value.fields
+
+        if (isEntryPoint) {
+          remoteJoinerQuery.alias = key
+          isEntryPoint = false
+        } else {
+          remoteJoinerQuery.expands!.push(expandObj)
+        }
+
+        extractRecursive(value, entityName)
```
Comment (adrien2p) [inline packages/orchestration/src/joiner/helpers.ts:50]: comment: based on previous comment if accepted, pass isEntryPoint as the third arg

**Review by adrien2p — APPROVED** [review]
LGTM, maybe just adding a unit test would be nice

**Comment by olivermrbl** [comment]
@adrien2p @carlos-r-l-rodrigues – I don't think I have fully understood the purpose of this helper. 

Can you add some more detail to it and perhaps provide a couple of ~~examples~~ scenarios?

**Comment by adrien2p** [comment]
At the moment, when you use the remote query, you can pass a RemoteJoinerQuery or a strong (gql). The RemoteJoinerQuery is not that simple to use and contains some properties that can be auto populated. This util aims to simplify the object to pass and take care of building the appropriate RemoteJoinerQuery

**Comment by olivermrbl** [comment]
> At the moment, when you use the remote query, you can pass a RemoteJoinerQuery or a strong (gql). The RemoteJoinerQuery is not that simple to use and contains some properties that can be auto populated. This util aims to simplify the object to pass and take care of building the appropriate RemoteJoinerQuery

This helps, thanks. Will we need to use this anywhere now?

**Comment by adrien2p** [comment]
> > At the moment, when you use the remote query, you can pass a RemoteJoinerQuery or a strong (gql). The RemoteJoinerQuery is not that simple to use and contains some properties that can be auto populated. This util aims to simplify the object to pass and take care of building the appropriate RemoteJoinerQuery
> 
> This helps, thanks. Will we need to use this anywhere now?

We will need to see, but might be easier to build an object with filters etc than a gql string. specially with complex filters and special filters

**Pull request merged**